### PR TITLE
MRG: Add support for fNIRS NIRStar v15.0 files

### PR DESCRIPTION
**Pull request by rob-luke**
#### What does this implement/fix?
Adds support for reading NIRX NIRStar version 15.0 files.

#### Additional information
This PR is dependent on https://github.com/mne-tools/mne-testing-data/pull/56 being merged. Then the testing version needs to be incremented in this PR.

### CODE REVIEW

File: mne/io/nirx/nirx.py [hunk 1]
```diff
@@ -116,8 +116,9 @@ def __init__(self, fname, preload=False, verbose=None):
         hdr.read_string(hdr_str)
 
         # Check that the file format version is supported
-        if hdr['GeneralInfo']['NIRStar'] != "\"15.2\"":
-            raise RuntimeError('Only NIRStar version 15.2 is supported')
+        if not any(item == hdr['GeneralInfo']['NIRStar'] for item in
+                   ["\"15.0\"", "\"15.2\""]):
+            raise RuntimeError('MNE does not support this NIRStar version')
```
Comment (larsoner) [inline mne/io/nirx/nirx.py]:
```suggestion
            raise RuntimeError('MNE does not support this NIRStar version'
                               ' (%s)' % (hdr['GeneralInfo']['NIRStar'],))
```

File: mne/io/nirx/tests/test_nirx.py [hunk 2]
```diff
@@ -137,6 +139,52 @@ def test_nirx_15_2():
...
+
+    # Test distance between optodes matches values from
+    allowed_distance_error = 0.0002
+    distances = raw._probe_distances()
+    assert_allclose(distances[::2], [
+        0.0301, 0.0315, 0.0343, 0.0368, 0.0408,
+        0.0399, 0.0393, 0.0367, 0.0336, 0.0447], atol=allowed_distance_error)
+
+
 @requires_testing_data
```
Comment (larsoner) [inline mne/io/nirx/tests/test_nirx.py]: Should add a `parametrize` over all three testing files here.

Comment (larsoner) [reply mne/io/nirx/tests/test_nirx.py]: Oh wait I'm doing that over in #6982, which hopefully I can fix and merge tomorrow, so better to hold off for now unless you want a conflict

Comment (rob-luke) [reply mne/io/nirx/tests/test_nirx.py]: Yep was planning to wait for #6982 before asking for review, sorry I should have been more specific about that.

Comment (larsoner) [reply mne/io/nirx/tests/test_nirx.py]: Okay #6982 is in, feel free to rebase and add it to the `parametrize`

**Comment by rob-luke** [comment]
@larsoner and @agramfort this is now ready for review/merge. Thanks

File: mne/io/nirx/nirx.py [hunk 3]
```diff
@@ -223,11 +229,12 @@ def prepend(list, str):
             # Store channel location
             # Channel locations for short channels are bodged,
             # for short channels use the source location and add small offset
-            if (has_short > 0) & (len(np.where(short_det == det + 1)[0]) > 0):
-                info['chs'][ch_idx2 * 2]['loc'][:3] = src_locs[src, :]
-                info['chs'][ch_idx2 * 2 + 1]['loc'][:3] = src_locs[src, :]
-                info['chs'][ch_idx2 * 2]['loc'][0] += 0.8
-                info['chs'][ch_idx2 * 2 + 1]['loc'][0] += 0.8
+            if has_short > 0:
+                if len(np.where(short_det == det + 1)[0]) > 0:
```
Comment (larsoner) [inline mne/io/nirx/nirx.py]: This is the same as `if det + 1 in short_det` right? If so it's clearer that it's just checking for membership.

Is this doing the right thing when `has_short > 0` and `det + 1 not in short_det` (i.e., don't adjust anything, as the `else` clause below isn't triggered either)?

File: mne/io/nirx/nirx.py [hunk 4]
```diff
@@ -133,11 +135,15 @@ def __init__(self, fname, preload=False, verbose=None):
                                 hdr['DataStructure']['S-D-Key'])], int)
 
         # Determine if short channels are present and on which detectors
-        has_short = np.array(hdr['ImagingParameters']['ShortBundles'], int)
-        short_det = [int(s) for s in
-                     re.findall(r'(\d+)',
-                     hdr['ImagingParameters']['ShortDetIndex'])]
-        short_det = np.array(short_det, int)
+        if any(key == 'shortbundles' for key in hdr['ImagingParameters']):
```
Comment (larsoner) [inline mne/io/nirx/nirx.py]: Can this just be `if 'shortbundles' in hdr['ImagingParameters']`? If `hdr` is dict-like then it should work

**Comment by rob-luke** [comment]
I also rethought the way short channels are positioned, and am now using the source location. I think this makes most sense, as the short channels are positioned as close to sources as possible.

**Comment by larsoner** [comment]
Nice, looks more readable to me now. Thanks @rob-luke !

**Pull request merged**